### PR TITLE
feat: added Authorization

### PR DESCRIPTION
**Pull request by srv-pranjal**
# Added Authorization

## Description

 - Added firebase for authorization
 - Added react-toastify for toast
 - Added auth context
 - Added Login, Signup, Profile and Logout
 - Added private routes
 
 ## Deploy preview link - https://quizzers-git-auth-srv-pranjal.vercel.app/

### CODE REVIEW

File: src/components/Navbar/Navbar.css [hunk 1]
```diff
@@ -27,12 +27,19 @@
   margin-right: 0.5rem;
 }
 
-.login-btn:hover {
+.nav .login-btn:hover {
   outline: 0.1rem solid var(--primary-color);
   background-color: unset;
+  color: var(--white-color) ;
```
Comment (ashishkatiyar2404) [inline src/components/Navbar/Navbar.css:33]: appreciation :  Good to see u take care of color variables

File: src/pages/Auth/Auth.css [hunk 2]
```diff
@@ -0,0 +1,151 @@
+.authentication {
```
Comment (ashishkatiyar2404) [inline src/pages/Auth/Auth.css:1]: appreciation: CSS code looks really nice

**Review by ashishkatiyar2404 — APPROVED** [review]
The animations are nice.
loader is also good,  and UI is also awesome.
You can merge now and work on other functionality

File: src/components/CategoryCard/CategoryCard.js [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
   const { quizDispatch } = useQuiz();
+  const { isLoggedIn } = useAuth();
   return (
     <Link
       to="/rules"
-      onClick={() =>
-        quizDispatch({ type: "SET_CATEGORY", payload: categoryName })
-      }
+      onClick={() => {
+        quizDispatch({ type: "SET_CATEGORY", payload: categoryName });
```
Comment (Namrata-J) [inline src/components/CategoryCard/CategoryCard.js:14]: suggestion: use constants instead of hardcodig type everytime!!

**Review by Namrata-J — APPROVED** [review]
- code is very much modular and structured. Have given a suggestion, you could try looking into it.
- also good commit messages used!!

**Pull request merged**